File: Spoj/main/FACEFRIEND.py
dF=[]
fof=[]
added=[]
count=0
for x in range(input()):
    a=raw_input().split()
    dF.append(a[0])
    for i in range(2,int(a[1])+2):fof.append(a[i])
    #print fof
for frnd in fof:
    if frnd not in dF and frnd not in added:
        count+=1
        added.append(frnd)
print count 

File: Spoj/main/FACEFRIEND.cpp
using namespace std;
#include<iostream>
#include<cstdio>
#include<cstdlib>
#include<map>

#define direct 1
#define fof 2
int main()
{         
	map <int,int> frnd;
	int n,m,temp,count=0;
	scanf("%d",&n);
	while(n--){
		scanf("%d",&temp);
		if(frnd[temp]==fof)count--;
		frnd[temp]=direct;
		scanf("%d",&m);
		while(m--){
			scanf("%d",&temp);
			if(frnd.find(temp)==frnd.end()){
				count++;
				frnd[temp]=fof;
			}
		}
	}
	printf("%d\n",count);
	return 0;
}
